Makefile

```diff
@@ -0,0 +1,49 @@
+pip:
+	pip install -e .
+
+pip-development: pip
+	pip install -e.[development]
+
+pip-docs: pip-development
+	pip install -r docs/requirements.txt
+	pip install doc2dash
+
+pip-wheel:
+	pip install wheel
+
+bdist-wheel:
+	python setup.py sdist bdist_wheel
+
+docs:
+	rm -rf docs/pyteal.docset
+	cd docs && \
+	make html && \
+	doc2dash --name pyteal --index-page index.html --online-redirect-url https://pyteal.readthedocs.io/en/ _build/html && \
+	tar -czvf pyteal.docset.tar.gz pyteal.docset
+
+build:
+	python -m scripts.generate_init --check
+
+ALLPY = docs examples pyteal scripts tests *.py
+black:
+	black --check $(ALLPY)
```

Comment:
Why `$(ALLPY)` and not `.`?

Reply:
It was going into my virutal env directory and failing because of that, so I wanted to explicitly define what is being checked. It's also handy to use this for the other linters. So I would prefer to keep this way. 

Another approach is to specifically exclude certain folders with `--exclude`, but then everyone would have to configure that specifically for their local excludes. 

But admittedly, the downside is that if you add another top-level directory, we wouldn't be formatting it by default. 

I don't feel strongly about this. LMK